test/plug/static_test.exs

```diff
@@ -843,4 +866,16 @@ defmodule Plug.StaticTest do
 
     assert conn.status == 200
   end
+
+  test "init/1 preserves a two-tuple :from with an application name and directory" do
+    assert %{from: {:foo, "foo/bar/foobar"}} =
+             Plug.Static.init(
+               at: "/",
+               from: {:foo, "foo/bar/foobar"}
+             )
+  end
+
+  test "init/1 transforms an application name :from to a two-tuple :from with the application name and a directory" do
+    assert %{from: {:foo, "priv/static"}} = Plug.Static.init(at: "/", from: :foo)
+  end
```

Comment:
Thank you. I am not sure those tests are useful, because they are pretty much testing internals. However, I assume you wrote them to get full coverage, is that correct?

Reply:
I agree with José, `init/1` is a callback that only makes sense together with `call/2`. Testing what `init/1` returns is not important because anyway the important part is that then `call/2` can accept what `init/1` returns. So this is definitely testing internals IMO.

Reply:
Thanks for the feedback.

Is there a resource that could help me understand this approach to testing?

Reply:
I don't have anything off the top of my head but searching for "black-box testing" for example could lead to some good resources.

Reply:
Does the remaining test also come under the above? I'm happy to close this PR if so :)